### ISSUE TITLE
Guide Modals - Star selection flow

### ISSUE THREAD

**Issue by MoeNick** [GeneralMagicio/pairwise-rf5#12]
Design: https://www.figma.com/design/WTT6ygvEonijX7Lc0pj6Ru/Pairwise-Desktop-App?node-id=1027-26613&t=b1CyiM3vB29UoGyh-4

- When the user for the first time clicks on a star, we can show this modal: 

![Image](https://github.com/user-attachments/assets/1ade0086-d486-4666-9ed6-a95fdc1cf3d9)

- If the user ONLY puts a star on another project on that screen we can show this modal:

![Image](https://github.com/user-attachments/assets/7438f875-5ba6-4080-b4ee-ec7083184e39)

- in the above scenario if the user selects a project we don't show the second modal again. 
- if the user undoes a star, we don't show the modals again. 
- It's only one time for he user in the whole game.

**Comment by Meriem-BM** [comment]
> Apologies @Meriem-BM But the design team has changed the first modal in the last minute, can I ask you to please replace it? https://www.figma.com/design/WTT6ygvEonijX7Lc0pj6Ru/Pairwise-Desktop-App?node-id=1027-24684&t=Fj5lgNTzetORwRdu-4 ![Image](https://private-user-images.githubusercontent.com/80677512/370681286-5d300852-e3b4-4bfc-8015-35026f2a38a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyNjYzOTUsIm5iZiI6MTcyNzI2NjA5NSwicGF0aCI6Ii84MDY3NzUxMi8zNzA2ODEyODYtNWQzMDA4NTItZTNiNC00YmZjLTgwMTUtMzUwMjZmMmEzOGE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI1VDEyMDgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyZmQ2ZmFkOGUzYzQ4YTI3NTViZWJkNmVhY2I0ZTcyYjQ4YTQyNmJiN2MyMzE0YmViZDY4ZjFjNGM4ZjhiNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7K0ClvY32yJz8PdY8d3tvZ0b6peloWdJDuu9dixvvJM)

@MoeNick, @maryjaf it's now on staging.

**Comment by maryjaf** [comment]
Thanks @Meriem-BM the new design has been added,
but one more thing, if I change the stars for the first time in for example  second or third or.. comparison , these modal aren't shown although  this is the first time thai I want to change the stars
actually they are shown only in first comparison    


https://github.com/user-attachments/assets/acd8fa05-f39a-4a85-9c30-41388802672a

**Comment by Meriem-BM** [comment]
> Thanks @Meriem-BM the new design has been added, but one more thing, if I change the stars for the first time in for example second or third or.. comparison , these modal aren't shown although this is the first time thai I want to change the stars actually they are shown only in first comparison
> 

cc: @MoeNick

**Comment by maryjaf** [comment]
> Thanks [@Meriem-BM](https://github.com/Meriem-BM) the new design has been added, but one more thing, if I change the stars for the first time in for example second or third or.. comparison , these modal aren't shown although this is the first time thai I want to change the stars actually they are shown only in first comparison
> 
>  Screen.Recording.2024-09-25.at.4.09.52.PM.mov

@MoeNick It should be fixed

**Comment by maryjaf** [comment]
> Thanks [@Meriem-BM](https://github.com/Meriem-BM) the new design has been added, but one more thing, if I change the stars for the first time in for example second or third or.. comparison , these modal aren't shown although this is the first time thai I want to change the stars actually they are shown only in first comparison
> 
>  Screen.Recording.2024-09-25.at.4.09.52.PM.mov

@MoeNick said It should be fixed

**Comment by Meriem-BM** [comment]
> [@MoeNick](https://github.com/MoeNick) said It should be fixed

Can you please retest this @maryjaf

**Comment by maryjaf** [comment]
It has been fixed but there is an ui issue, please take a look on below screen record @Meriem-BM 

Uploading Screen Recording 2024-09-27 at 4.47.02 PM.mov…

**Comment by Meriem-BM** [comment]
> It has been fixed but there is an ui issue, please take a look on below screen record @Meriem-BM
> 
> Uploading Screen Recording 2024-09-27 at 4.47.02 PM.mov…

I can see the record @maryjaf, don't know if it's uploading

**Comment by maryjaf** [comment]
> > It has been fixed but there is an ui issue, please take a look on below screen record @Meriem-BM
> > Uploading Screen Recording 2024-09-27 at 4.47.02 PM.mov…
> 
> I can see the record @maryjaf, don't know if it's uploading

I've uploaded it again

**Comment by maryjaf** [comment]
There is another problem, if without closing first modal, I tap on stars of other card, the below problem happens


https://github.com/user-attachments/assets/71ba1582-2a74-4df2-bc1a-3d0133c496e2

@Meriem-BM

**Comment by Meriem-BM** [comment]
> There is another problem, if without closing first modal, I tap on stars of other card, the below problem happens
> 
>  Screen.Recording.2024-09-28.at.6.53.16.PM.mov 
> @Meriem-BM

I made a fix for this one, if you can test @maryjaf

**Comment by maryjaf** [comment]
> > There is another problem, if without closing first modal, I tap on stars of other card, the below problem happens
> > Screen.Recording.2024-09-28.at.6.53.16.PM.mov
> > @Meriem-BM
> 
> I made a fix for this one, if you can test @maryjaf

Thanks @Meriem-BM it has been fixed but this tiny ui issue in guide modal hasn't been fixed yet

> It has been fixed but there is an ui issue, please take a look on below screen record @Meriem-BM
> 
>  Screen.Recording.2024-09-27.at.4.47.02.PM.mov

**Comment by maryjaf** [comment]
https://github.com/user-attachments/assets/e5f991cd-687d-43a4-a974-3bfed1204503

**Comment by Meriem-BM** [comment]
> Screen.Recording.2024-09-29.at.4.47.48.PM.mov

I made the fix if you can check @maryjaf

**Comment by maryjaf** [comment]
> > Screen.Recording.2024-09-29.at.4.47.48.PM.mov
> 
> I made the fix if you can check [@maryjaf](https://github.com/maryjaf)

Thanks @Meriem-BM It has been fixed
![Image](https://github.com/user-attachments/assets/f74fbae2-e574-4324-9068-203e59abbfcc)

**Comment by maryjaf** [comment]
I found new problem could you please take a look @Meriem-BM 

imagin in the first selection of stars you set some stars for a project , the first guide modal is shown correctly but you don't set star for another project , in this case in next steps if you set star for a project, the second guide modal will be shown

**Comment by maryjaf** [comment]
> I found new problem could you please take a look [@Meriem-BM](https://github.com/Meriem-BM)
> 
> imagin in the first selection of stars you set some stars for a project , the first guide modal is shown correctly but you don't set star for another project , in this case in next steps if you set star for a project, the second guide modal will be shown

maybe it's my mistake , i'll check again

**Comment by maryjaf** [comment]
> I found new problem could you please take a look [@Meriem-BM](https://github.com/Meriem-BM)
> 
> imagin in the first selection of stars you set some stars for a project , the first guide modal is shown correctly but you don't set star for another project , in this case in next steps if you set star for a project, the second guide modal will be shown

https://github.com/user-attachments/assets/8989b6c2-4423-4c98-9559-48482603de97

@Meriem-BM

**Comment by Meriem-BM** [comment]
> > I found new problem could you please take a look [@Meriem-BM](https://github.com/Meriem-BM)
> > imagin in the first selection of stars you set some stars for a project , the first guide modal is shown correctly but you don't set star for another project , in this case in next steps if you set star for a project, the second guide modal will be shown
> 
>  Screen.Recording.2024-09-30.at.4.54.51.PM.mov 
> @Meriem-BM

This behavior is expected, as I ensure the 2nd modal appears when two projects are rated for the first time, even if the 1st modal has been shown in previous comparisons. I can adjust it so that once the 1st modal appears and the user selects a project, neither modal will be shown again.

**Comment by MoeNick** [comment]
Yes please @Meriem-BM we should limit user only see both in single comparison. If user passes that screen while seeing the first one. , we NEVERshould show up the second modal again. 
![image](https://github.com/user-attachments/assets/702cafbc-6c19-4aeb-b296-1cfba0de4e59)

**Comment by Meriem-BM** [comment]
Just made the fix, it's on staging if you can test @maryjaf

**Comment by maryjaf** [comment]
> but in this screen the first modal is shown for me correctly, since i didn't set any star for second project, the second modal isn't shown as expected but in next step by setting star for second project this modal is shown !

Thanks @Meriem-BM it has been fixed

**Comment by MoeNick** [comment]
van you merge please @Meriem-BM ?

**Comment by Meriem-BM** [comment]
> van you merge please @Meriem-BM ?

Yeah, I think it's merged to prod by Mahdi
